admin/pre_reg/views.py

```diff
@@ -20,169 +16,181 @@
-    except (TypeError, ValueError):
-        page_number = 1
-
-    page = paginator.page(page_number)
-
-    try:
-        drafts = [serializers.serialize_draft_registration(d, json_safe=False) for d in page]
-    except EmptyPage:
-        drafts = []
-
-    for draft in drafts:
-        draft['form'] = DraftRegistrationForm(draft)
-
-    context = {
-        'drafts': drafts,
-        'page': page,
-        'IMMEDIATE': serializers.IMMEDIATE,
-    }
-    return render(request, 'pre_reg/prereg.html', context)
-
-
-@user_passes_test(is_in_prereg_group)
-def view_draft(request, draft_pk):
-    """Redirects to prereg form review page if user has prereg access
-    :param draft_pk: Unique id for selected draft
-    :return: Redirect to prereg form review page with draft obj
-    """
-    draft = get_draft_or_error(draft_pk)
-    context = {
-        'draft': serializers.serialize_draft_registration(draft)
-    }
-    return render(request, 'pre_reg/edit_draft_registration.html', context)
-
-
-@user_passes_test(is_in_prereg_group)
-def view_file(request, node_id, provider, file_id):
-    file = FileNode.load(file_id)
-    wb_url = file.generate_waterbutler_url()
-    return redirect(wb_url)
-
+from website.project.model import DraftRegistration
+
+from admin.base.utils import PreregAdmin
+
+
+class DraftListView(PreregAdmin, ListView):
+    template_name = 'pre_reg/draft_list.html'
+    ordering = '-date'
+    context_object_name = 'draft'
+
+    def get_queryset(self):
+        query = build_query(self.request.GET.get('status', 'all'))
+        ordering = self.get_ordering()
+        if 'initiator' in ordering:
+            return DraftRegistration.find(query).sort(ordering)
+        return sort_drafts(DraftRegistration.find(query), ordering)
+
+    def get_context_data(self, **kwargs):
+        query_set = kwargs.pop('object_list', self.object_list)
+        page_size = self.get_paginate_by(query_set)
+        paginator, page, query_set, is_paginated = self.paginate_queryset(
+            query_set, page_size)
+        return {
+            'drafts': [
+                serializers.serialize_draft_registration(d, json_safe=False)
+                for d in query_set
+            ],
+            'page': page,
+            'p': self.get_paginate_by(query_set),
+            'SORT_BY': SORT_BY,
+            'order': self.get_ordering(),
+            'VIEW_STATUS': VIEW_STATUS,
+            'status': self.request.GET.get('status', 'all')
+        }
+
+    def get_paginate_by(self, queryset):
+        return int(self.request.GET.get('p', 10))
+
+    def get_paginate_orphans(self):
+        return int(self.get_paginate_by(None) / 11.0) + 1
+
+    def get_ordering(self):
+        return self.request.GET.get('order_by', self.ordering)
+
+
+class DraftDetailView(PreregAdmin, DetailView):
+    template_name = 'pre_reg/draft_detail.html'
+    context_object_name = 'draft'
+
+    def get(self, request, *args, **kwargs):
```

Comment:
Looking at the source of the (built-in django) parent classes almost always gives a good pattern for common operations.

get: https://ccbv.co.uk/projects/Django/1.9/django.views.generic.detail/DetailView/#get
get_object: https://ccbv.co.uk/projects/Django/1.9/django.views.generic.detail/DetailView/#get_object